src/api/helpers.jl

```diff
@@ -361,18 +361,26 @@ function h5p_get_efile_prefix(plist)
 end
 
 function h5p_get_external(plist, idx = 0)
-    name_size = Csize_t(1024)
-    name = zeros(UInt8, name_size)
-    offset = Ref{off_t}(0)
+    offset = Ref{Int64}(0)
     sz = Ref{hsize_t}(0)
-    h5p_get_external(plist, idx, name_size, name, offset, sz)
-    # name may not be null terminated according to H5P_GET_EXTERNAL documentation
-    nul_idx =  findfirst(==(0x00), name)
-    name_size = nul_idx === nothing ? name_size : nul_idx-1
+    name_size = 64
+    name = Base.StringVector(name_size)
+    null_id = nothing
+    while true
+        h5p_get_external(plist, idx, name_size+1, name, offset, sz)
+        null_id = findfirst(==(0x00), name)
+        if isnothing(null_id)
+            name_size *= 2
+            name = Base.StringVector(name_size)
+        else
+            resize!(name, null_id-1)
+            break
+        end
+    end
     @static if Sys.iswindows() && sizeof(Int) == 4
         sz[] &= 0xffffffff
     end
```

Comment:
@mkitti  could you explain why this is necessary?

Reply:
If we take that out, I think this fails on 32-bit Windows. I'm guessing that one of our types is coded wrong for 32-bit windows.

Reply:
Oddly enough, however, the type is  hsize_t here: https://github.com/HDFGroup/hdf5/blob/78fe6751b5a682111e8bc2bf7ce944c55d52ba56/src/H5Pdcpl.c#L2974

Reply:
I'm thinking to change the API signature to UInt, instead of this hack.

Reply:
Is our `hsize_t` definition wrong?
https://github.com/JuliaIO/HDF5.jl/blob/ad2f86d38d0298de7e187d7e09bb26e4312fb562/src/api/types.jl#L9

Reply:
From the source code of libhdf5 that is the correct definition. It's a `typedef unsigned long long`

It's just that this method `h5p_get_external ` has issues on win32 with  `sz`  being UInt64 on a 64-bit windows. Changing 
```
@bind h5p_get_external(plist::hid_t, idx::Cuint, name_size::Csize_t, name::Ptr{Cuchar}, offset::Ptr{off_t}, size::Ptr{hsize_t})::herr_t "Error getting external file properties"
```
to
```
@bind h5p_get_external(plist::hid_t, idx::Cuint, name_size::Csize_t, name::Ptr{Cuchar}, offset::Ptr{off_t}, size::Ptr{UInt})::herr_t "Error getting external file properties"
```
fixes the problem. Despite the fact that the libhdf5 is using `hsize_t`  😕

Reply:
This is probably a bug in HDF5 on Win32. Frankly, it still looks like a hack to me because I think 32-bit windows is the only 32-bit platform we test on.

Reply:
Well your hack looks slightly better than my hack, so might as well commit it.

Reply:
I'd like to double check on 32-bit Linux, I think we have the capability to test it, but I probably disabled it due to redundancy with the win32 system.

Reply:
We don't have a HDF5_jll to test it on.

Reply:
welp....

In any case, I don't think it hurts too much to use `UInt` for this entry anyways... The external file data would have to bigger than `UInt32` for this to be a problem...

Reply:
Did we find a configuration that will pass CI?

Reply:
I think so, I just haven't gotten around to changing the PR.